server/game/cards/characters/03/aryastark.js

```diff
@@ -0,0 +1,36 @@
+const DrawCard = require('../../../drawcard.js');
+
+class AryaStark extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                onCharacterKilled: (event, player, card) => (
+                    this.controller === card.controller &&
+                    card.getFaction() === this.getFaction())
+            },
+            cost: ability.costs.sacrificeSelf(),
+            handler: () => {
+                this.game.promptForSelect(this.controller, {
+                    cardCondition: card => (
+                        card.location === 'play area' && 
+                        card.getType() === 'character' &&
+                        card.getStrength() <= 3),
+                    activePromptTitle: 'Select a character',
+                    waitingPromptTitle: 'Waiting for opponent to use ' + this.name,
+                    onSelect: (player, card) => this.onCardSelected(player, card)
+                });
+            }
+        });
+    }
+
+    onCardSelected(player, card) {
+        card.controller.killCard(card);
```

Comment:
Sorry, missed this in initial review - should be `killCharacter` not `killCard`.